src/include/mpir_mem.h

```diff
@@ -78,9 +78,105 @@ extern "C" {
 /* Define the string copy and duplication functions */
 /* ------------------------------------------------------------------------- */
 
-#define MPIR_Memcpy(dst, src, len)              \
+#include "string.h"
+
+#ifdef HAVE_MM256_STREAM_SI256
+#include <immintrin.h>
+
+static inline void MPIR_Memcpy_stream(void *dest, const void *src, size_t n)
```

Comment:
To clarify, `MPL_memcpy_stream` in my mind is fine. It is a context-free utility function that belongs in MPL. I was suggesting to provide a MPIR wrapper just like `MPIR_Memcpy` is a wrapper for `memcpy`.

Reply:
I see. That's fine too.

Reply:
I've made that change. It's in a new fixup commit here.

Reply:
I think it looks good!

Reply:
I think so too. Squash the commits and clear the tests and I think this PR is ready